app/assets/src/components/views/SampleUploadFlow/UploadProgressModal.jsx

```diff
@@ -0,0 +1,486 @@
+  };
+
+  someLocalSamplesInProgress = () => {
+    const { samples } = this.props;
+    const { sampleUploadStatuses } = this.state;
+    return some(
+      sample => sampleUploadStatuses[sample.name] === undefined,
+      samples
+    );
+  };
+
+  someLocalSamplesFailed = () => {
+    const { samples } = this.props;
+    const { sampleUploadStatuses } = this.state;
+    return some(
+      sample => sampleUploadStatuses[sample.name] === "error",
+      samples
+    );
+  };
+
+  renderSampleStatus = sample => {
+    const { sampleUploadStatuses } = this.state;
+
+    if (sampleUploadStatuses[sample.name] === "error") {
+      return (
+        <React.Fragment>
+          <AlertIcon className={cs.alertIcon} />
+          Upload failed.
+        </React.Fragment>
+      );
+    }
+
+    if (sampleUploadStatuses[sample.name] === "success") {
+      return (
+        <React.Fragment>
+          <CheckmarkIcon className={cs.checkmarkIcon} />
+          Uploaded and sent to pipeline
+        </React.Fragment>
+      );
+    }
+
+    const uploadPercentage = this.getSampleUploadPercentage(sample);
+    if (uploadPercentage === undefined) {
+      return "Waiting to upload...";
+    }
+
+    const totalSize = this.getSampleTotalSize(sample);
+
+    return `Uploaded ${formatFileSize(
+      totalSize * uploadPercentage
+    )} of ${formatFileSize(totalSize)}...`;
+  };
+
+  renderFailedSamplesTitle = () => (
+    <React.Fragment>
+      <div className={cs.titleWithIcon}>
+        <AlertIcon className={cs.alertIcon} />
+        Some samples failed to upload
+      </div>
+      <div className={cs.subtitle}>
+        <a
+          className={cs.helpLink}
+          href="mailto:help@idseq.net"
+          onClick={() =>
+            logAnalyticsEvent("UploadProgressModal_contact-us-link_clicked")
+          }
+        >
+          Contact us for help.
+        </a>
+      </div>
+    </React.Fragment>
+  );
+
+  renderTitle = () => {
+    const { samples, project, uploadType } = this.props;
+    const { uploadComplete, failedSampleNames } = this.state;
+
+    if (uploadType === "remote" || uploadType === "basespace") {
+      if (uploadComplete) {
+        // If any samples failed.
+        if (failedSampleNames.length > 0) {
+          return this.renderFailedSamplesTitle();
+          // If all samples succeeded.
+        } else {
+          return (
+            <React.Fragment>
+              <div className={cs.titleWithIcon}>
+                <CheckmarkIcon className={cs.checkmarkIcon} />
+                {samples.length} samples successfully created.
+              </div>
+              <div className={cs.instructions}>
+                Our servers have started uploading your sample files from{" "}
+                {uploadType === "basespace" ? "Basespace" : "S3"}. After the
+                upload is complete, your samples will automatically start
+                processing. Please check back soon.
```

Comment:
I worry that this text is getting too long. I'll remove the `Please check back soon` for now since it might be misinterpreted as asking the user to stay on the upload page. I think it's okay to leave that part off...with our training and docs it should hopefully be clear where to view their samples.